akka-actor/src/main/scala/akka/actor/dungeon/FaultHandling.scala

```diff
@@ -240,10 +240,10 @@ private[akka] trait FaultHandling { this: ActorCell =>
     try if (a ne null) a.aroundPostStop()
     catch handleNonFatalOrInterruptedException { e =>
       publish(Error(e, self.path.toString, clazz(a), e.getMessage))
-    } finally try dispatcher.detach(this)
+    } finally try stopFunctionRefs()
+    finally try dispatcher.detach(this)
```

Comment:
dispatcher.detach will close the mailbox, which will send queued messages to deadLetters, and the message adapter FunctionRef was not stopped before that. Without async boundaries that caused infinite loop.